flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/node/MethodCall.java

```diff
@@ -198,35 +199,41 @@ private void setCustomProperties(Collection<ParameterResult> functionParameters)
             customPropBuilder
                     .metadata()
-                    .label(unescapedParamName)
-                    .description(paramResult.description())
-                    .stepOut()
+                        .label(unescapedParamName)
+                        .description(paramResult.description())
+                        .stepOut()
                     .codedata()
-                    .kind(paramResult.kind().name())
-                    .originalName(paramResult.name())
-                    .importStatements(paramResult.importStatements())
-                    .stepOut()
+                        .kind(paramResult.kind().name())
+                        .originalName(paramResult.name())
+                        .importStatements(paramResult.importStatements())
+                        .stepOut()
                     .placeholder(paramResult.defaultValue())
                     .typeConstraint(paramResult.type())
                     .editable()
                     .defaultable(paramResult.optional() == 1);
 
-            if (paramResult.kind() == Parameter.Kind.INCLUDED_RECORD_REST) {
-                if (hasOnlyRestParams) {
-                    customPropBuilder.defaultable(false);
-                }
-                unescapedParamName = "additionalValues";
-                customPropBuilder.type(Property.ValueType.MAPPING_EXPRESSION_SET);
-            } else if (paramResult.kind() == Parameter.Kind.REST_PARAMETER) {
-                if (hasOnlyRestParams) {
-                    customPropBuilder.defaultable(false);
-                }
-                customPropBuilder.type(Property.ValueType.EXPRESSION_SET);
-            } else if (paramResult.kind() == Parameter.Kind.REQUIRED) {
-                customPropBuilder.type(Property.ValueType.EXPRESSION).value(paramResult.defaultValue());
-            } else {
-                customPropBuilder.type(Property.ValueType.EXPRESSION);
+            switch (paramResult.kind()) {
+                case INCLUDED_RECORD_REST:
+                    if (hasOnlyRestParams) {
+                        customPropBuilder.defaultable(false);
+                    }
+                    unescapedParamName = "additionalValues";
+                    customPropBuilder.type(Property.ValueType.MAPPING_EXPRESSION_SET);
+                    break;
+                case REST_PARAMETER:
+                    if (hasOnlyRestParams) {
+                        customPropBuilder.defaultable(false);
+                    }
+                    customPropBuilder.type(Property.ValueType.EXPRESSION_SET);
+                    break;
+                case REQUIRED:
+                    customPropBuilder.type(Property.ValueType.EXPRESSION).value(paramResult.defaultValue());
+                    break;
+                default:
+                    customPropBuilder.type(Property.ValueType.EXPRESSION);
+                    break;
```

Comment:
You can use an [enhanced switch statement](https://www.geeksforgeeks.org/enhancements-for-switch-statement-in-java-13/) in here instead.

Reply:
+1